src/main/java/kr/motd/maven/os/DetectExtension.java

```diff
@@ -100,8 +112,21 @@ public void afterProjectsRead(MavenSession session) throws MavenExecutionExcepti
         }
     }
 
+    /**
+     * Inspects the session's user and project properties for the {@link
+     * DetectMojo#CLASSIFIER_WITH_LIKES_PROPERTY} and separates the property into a list.
+     */
+    private static List<String> getClassifierWithLikes(MavenSession session) {
+        // Check to see if the project defined the
+        Properties props = new Properties();
+        props.putAll(session.getUserProperties());
+        props.putAll(session.getCurrentProject().getProperties());
+        return DetectMojo.getClassifierWithLikes(
+            props.getProperty(DetectMojo.CLASSIFIER_WITH_LIKES_PROPERTY));
+    }
+
     private void injectSession(MavenSession session, Map<String, String> dict) {
-        Properties sessionExecProps = session.getExecutionProperties();
+        Properties sessionExecProps = session.getSystemProperties();
```

Comment:
@trustin `session.getExecutionProperties()` is deprecated.  It's the union of `systemPropertes` and `userProperties`. Any reason you chose it over just `systemProperties`?